src/main/java/com/conveyal/datatools/manager/controllers/api/DeploymentController.java

```diff
@@ -449,9 +482,95 @@ private static String deploy (Request req, Response res) {
                 target);
             logMessageAndHalt(req, HttpStatus.ACCEPTED_202, message);
         }
+
+        // If pelias update is requested, launch pelias update job
+        if (deployment.peliasUpdate) {
+            updatePelias(req, res, job);
+        }
+
         return SparkUtils.formatJobMessage(job.jobId, "Deployment initiating.");
     }
 
+    /**
+     * Start an updatePelias job which will trigger the webhook, then check for status updates.
+     */
+    private static String updatePelias (Request req, Response res, DeployJob deployJob) {
+        // Check parameters supplied in request for validity.
+        Auth0UserProfile userProfile = req.attribute("user");
+        Deployment deployment = getDeploymentWithPermissions(req, res);
+        Project project = Persistence.projects.getById(deployment.projectId);
+        if (project == null) {
+            logMessageAndHalt(req, 400, "Internal reference error. Deployment's project ID is invalid");
+        }
+
+        // Get log upload URI from deploy job
+        AmazonS3URI logUploadS3URI = deployJob.getS3FolderURI();
+
+        // Execute the pelias update job and keep track of it
+        PeliasUpdateJob peliasUpdateJob = new PeliasUpdateJob(userProfile, "Updating Custom Geocoder Database", deployment, logUploadS3URI);
+        JobUtils.heavyExecutor.execute(peliasUpdateJob);
+        return SparkUtils.formatJobMessage(peliasUpdateJob.jobId, "Pelias update initiating.");
+    }
+
+    /**
+     * Uploads a file from Spark request object to the s3 bucket of the deployment the Pelias Update Job is associated with.
+     * Follows https://github.com/ibi-group/datatools-server/blob/dev/src/main/java/com/conveyal/datatools/editor/controllers/api/EditorController.java#L111
```

Comment:
I was thinking that we could take the opportunity to reuse some of the existing logic there. Perhaps some of the internals go into SparkUtils and/or S3Utils.

Reply:
Unified in 2abcd28674fc5e4185a71587c261094ec53700bd